.github/workflows/codeql-analysis.yml

```diff
@@ -15,19 +15,21 @@ name: "CodeQL"
 on:
   pull_request:
     branches:
-      - master
-      - opendistro-*
+      - main
   push:
-    branches:
-      - master
-      - opendistro-*
-
+    branches-ignore:
+      - 'dependabot/**'
+    paths:
+      - '**/*.java'
+      - '.github/workflows/codeql-analysis.yml'
 
 jobs:
   analyze:
     name: CodeQL-Scan
-    runs-on: ubuntu-18.04
-
+    runs-on: ubuntu-latest
```

Comment:
I think it's better to run tests vs a known version, rather than using latest.  This gives us fewer variables in our test framework, and prevents random failures when the product changes versions (and potentially breaks a test). 
Thoughts?

Reply:
This workflow never run before. `ubuntu-18.04` is deprecated already.
All other linux jobs run on `ubuntu-latest` too, so I put it here.

Reply:
Which version is `ubuntu-latest`?  My concern is that we aren't in control of the `ubuntu-latest` version, and it might cause errors spontaneously.  If we specify a version, it will eventually removed, but then we can just raise a new PR to update it correctly to a version that works.

Reply:
I agree. I will change version of `ubuntu` for all workflows.
https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources